pkg/kubelet/eviction/threshold_notifier_linux.go

```diff
@@ -90,28 +90,32 @@ func NewMemCGThresholdNotifier(path, attribute, threshold, description string, h
 	}, nil
 }
 
-func getThresholdEvents(eventfd int, eventCh chan<- int) {
+func getThresholdEvents(eventfd int, eventCh chan<- struct{}, stopCh <-chan struct{}) {
 	for {
 		buf := make([]byte, 8)
 		_, err := syscall.Read(eventfd, buf)
 		if err != nil {
 			return
 		}
-		eventCh <- 0
+
+		select {
+		case eventCh <- struct{}{}:
+		case <-stopCh:
+			return
+		}
 	}
 }
 
 func (n *memcgThresholdNotifier) Start(stopCh <-chan struct{}) {
-	eventCh := make(chan int, 1)
-	go getThresholdEvents(n.eventfd, eventCh)
+	eventCh := make(chan struct{}, 1)
```

Comment:
Actually, I don't think this channel should be buffered. With the buffer, this is what could happen:

1. stop channel is triggered, causing this function to exit
2. the go function `getThresholdEvents` reads an event (I'm assuming this syscall can block for a long time?)
3. the `eventCh` case is randomly selected in the select statement
4. it goes back to waiting for an event...

Without the buffer, the `eventCh` case will never be selected since there wouldn't be a receiver